Source/WebCore/editing/Editor.cpp

```diff
@@ -4326,6 +4326,9 @@ void Editor::cloneAttachmentData(const String& fromIdentifier, const String& toI
 
 void Editor::didInsertAttachmentElement(HTMLAttachmentElement& attachment)
 {
+    if (attachment.isImageOnly())
+        return;
```

Comment:
We still need to add the attachment to the document, because it's later needed when fetching the appropriate icon. I think the proposed patch is the best place to fix this at the moment.

Reply:
Looks like this cannot be done, since the document lookup is still required for rendering purposes, which is the entire point of the "image only" workaround.

The unnecessary editor state update is an unfortunate consequence of the current implementation.